pkg/kubectl/cmd/util/factory.go

```diff
@@ -176,7 +177,25 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			if cfg.GroupVersion != nil {
 				cmdApiVersion = *cfg.GroupVersion
 			}
-
+			client, err := clients.ClientForVersion(&unversioned.GroupVersion{Version: "v1"})
+			if err != nil {
+				fmt.Printf("Failed to create client: %v", err)
+				return nil, nil
+			}
+			versions, gvks, err := GetThirdPartyGroupVersions(client.RESTClient)
```

Comment:
Making this call needs to be optional in some way.  Perhaps a separate method on the factory that explicitly asks for a discovery based `RESTMapper` or more likely explicitly asks for one based on cache + compiled in knowledge?